tests/testsuite/member_discovery.rs

```diff
@@ -0,0 +1,36 @@
+//! Tests for workspace member discovery.
+
+//use cargo::core::resolver::ResolveError;
+use cargo::core::{Shell, Workspace};
+use cargo::ops;
+use cargo::util::config::Config;
+
+use cargo_test_support::install::cargo_home;
+use cargo_test_support::project;
+use cargo_test_support::registry;
+
+/// Tests exclusion of non-directory files from workspace member discovery using glob `*`.
+#[cargo_test]
+fn bad_file_member_exclusion() {
+    let p = project()
+        .file(
+            "Cargo.toml",
+            r#"
+            [workspace]
+            members = [ "crates/*" ]
+        "#,
+        )
+        .file("crates/.DS_Store", "PLACEHOLDER")
+        .build();
```

Comment:
Could this maybe add a real crate (`crates/foo/Cargo.toml` and `crates/foo/src/lib.rs`)?  It seems a little strange to test for the case where no members are found, since that is a bit unusual.  Checking that it found at least one member would also ensure that the test is exercising the appropriate code path.